masterslave.q

```diff
@@ -0,0 +1,102 @@
+//Master slave algorithm
+//Each process initially assumes it's the master process
+//Master process is determined by oldest process start time
+//Connect to other slave processes and retrieve details to determine which process is master
+
+//run lines
+//q torq.q -load masterslave.q -proctype lb -procname lb1 -p 6100 -debug -parentproctype wdb
+//q torq.q -load masterslave.q -proctype lb -procname lb2 -p 6101 -debug -parentproctype wdb
+
+//test cases
+//one process starts up, it becomes master
+//two processes, only one is master
+//two processes, master dies, slave takes over
+//two processes, one is made master, new process added, original master(if it's starttime
+// is still oldest)
+//two processes, one is made master, an older process is connected, recently connected process
+// should then become master
+//3 processes, one master, master dies, one of the remaining two takes over
+
+.servers.CONNECTIONS:distinct .servers.CONNECTIONS,.proc.proctype;
+.servers.startup[];
+
+\d .masterslave
+
+init:{
+ .masterslave.checkifmaster[];
+ .servers.connectcustom:{[f;x] .masterslave.addmember[x `.proc.procname];f@x}@[value;`.servers.connectcustom;{{}}];
+ .masterslave.masterupdate[]
+ };
+
+//table
+statustable:([handle:`int$()] procname:`$();starttimeUTC:`timestamp$();ismaster:`boolean$());
+
+//store own process start timestamp
+start:.z.p;
+
+//set details dict with own details
+details:{
+ `procname`starttimeUTC!(.proc.procname;.masterslave.start)
+ };
+
+//get details of procname provided
+getdetails:{[processname]
+ (first exec w from .servers.SERVERS where procname like raze string processname,not null w) (`.masterslave.details;[])
+ };
+
+//update .masterslave.statustable with other proc details and update ismaster col to determine which process is master
+addmember:{[processname]
+ `.masterslave.statustable upsert .masterslave.getdetails[processname],
+  (`handle`ismaster)!(first exec w from .servers.SERVERS  where procname like processname,not null w;1b)
```

Comment:
Is this return value used for anything?

Reply:
as in when it returns `.masterslave.statustable?
This func is just to upsert process details dictionary into the .ms.statustable, I'm not aware of any other return value?